src/components/follow/follow.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { Component } from 'react';
+import { PropTypes } from 'prop-types';
+import { connect } from 'react-redux';
+
+import { NavBar } from '../../components/common';
+import Button from '../common/button';
+import { isFollowing } from '../../helpers/followHelper';
+import {
+  followUser,
+  unfollowUser,
+  fetchFollowing,
+  fetchFollowers
+} from '../../actions/followAction';
+
+class Follow extends Component {
+  handleFollow = username => {
+    alert(`Are you sure to unfollow ${username}`);
+    this.props.unfollowUser(username);
+  };
+
+  componentDidUpdate(prevProps) {
+    const { isFollowed, isFollowingFetched } = this.props.follow;
+    if (isFollowed === 'done' || isFollowed === 'unfollowed') {
+      const { auth } = this.props;
+      if (isFollowingFetched !== 'done') {
+        this.props.fetchFollowing(auth.user.username);
+      } else {
+        this.props.fetchFollowers(auth.user.username);
+      }
+    }
+    let prevFollowers = prevProps.follow.isFollowersFetched;
+    let currentFollowers = this.props.follow.isFollowersFetched;
+
+    if (prevFollowers !== currentFollowers) {
+      let following = this.props.follow.following;
+      let followers = this.props.follow.follower;
+      let newLocationState = {
+        ...this.props.location.state,
+        following,
+        followers
+      };
+      this.props.location.state = newLocationState;
+      const { state } = this.props.location;
+      this.props.history.push({
+        pathname: '/follow',
+        state
+      });
+    }
+  }
+  render() {
+    const { following, followers } = this.props.location.state;
+    return (
+      <div>
+        <NavBar />
+        <div className='Follow'>
```

Comment:
Put a title on this page that indicates users where they are

Reply:
@AJAkimana. the titles are already there. there is one above followers and another above following.